tests/check/check.go

```diff
@@ -1206,45 +1225,136 @@ condition xcond(x: string) {
-								// TODO
-								TupleKeys: []*openfgav1.TupleKey{},
-							},
-							Context: assertion.Context,
-							Trace:   true,
-						})
-
-						if assertion.ErrorCode == 0 {
-							require.NoError(t, err, detailedInfo)
-							require.Equal(t, assertion.Expectation, resp.GetAllowed(), detailedInfo)
-						} else {
-							require.Error(t, err, detailedInfo)
-							e, ok := status.FromError(err)
-							require.True(t, ok, detailedInfo)
-							require.Equal(t, assertion.ErrorCode, int(e.Code()), detailedInfo)
-						}
+					t.Run(fmt.Sprintf("assertion_check_%s", assertion.Name), func(t *testing.T) {
+						assertCheck(ctx, t, assertion, stage, client, storeID, modelID)
+					})
+					t.Run(fmt.Sprintf("assertion_list_objects_%s", assertion.Name), func(t *testing.T) {
+						assertListObjects(ctx, t, assertion, client, storeID, modelID)
+					})
+					t.Run(fmt.Sprintf("assertion_list_users_%s", assertion.Name), func(t *testing.T) {
+						assertListUsers(ctx, t, assertion, client, storeID, modelID)
 					})
 				}
 			})
 		}
 	})
 }
 
+func assertCheck(ctx context.Context, t *testing.T, assertion *checktest.Assertion, stage *stage, client ClientInterface, storeID string, modelID string) {
+	detailedInfo := fmt.Sprintf("Check request: %s. Tuples: %s. Contextual tuples: %s", assertion.Tuple, stage.Tuples, assertion.ContextualTuples)
+
+	var tupleKey *openfgav1.CheckRequestTupleKey
+	if assertion.Tuple != nil {
+		tupleKey = &openfgav1.CheckRequestTupleKey{
+			User:     assertion.Tuple.GetUser(),
+			Relation: assertion.Tuple.GetRelation(),
+			Object:   assertion.Tuple.GetObject(),
+		}
+	}
+	resp, err := client.Check(ctx, &openfgav1.CheckRequest{
+		StoreId:              storeID,
+		AuthorizationModelId: modelID,
+		TupleKey:             tupleKey,
+		ContextualTuples: &openfgav1.ContextualTupleKeys{
+			// TODO
+			TupleKeys: []*openfgav1.TupleKey{},
+		},
+		Context: assertion.Context,
+		Trace:   true,
+	})
+
+	if assertion.ErrorCode == 0 {
+		require.NoError(t, err, detailedInfo)
+		require.Equal(t, assertion.Expectation, resp.GetAllowed(), detailedInfo)
+	} else {
+		require.Error(t, err, detailedInfo)
+		e, ok := status.FromError(err)
+		require.True(t, ok, detailedInfo)
+		require.Equal(t, assertion.ErrorCode, int(e.Code()), detailedInfo)
+	}
+}
+
+func assertListObjects(ctx context.Context, t *testing.T, assertion *checktest.Assertion, client ClientInterface, storeID string, modelID string) {
+	objectType, _ := tuple.SplitObject(assertion.Tuple.GetObject())
+	resp, err := client.ListObjects(ctx, &openfgav1.ListObjectsRequest{
+		StoreId:              storeID,
+		AuthorizationModelId: modelID,
+		Type:                 objectType,
+		User:                 assertion.Tuple.GetUser(),
+		Relation:             assertion.Tuple.GetRelation(),
+		ContextualTuples: &openfgav1.ContextualTupleKeys{
+			// TODO
+			TupleKeys: []*openfgav1.TupleKey{},
+		},
+		Context: assertion.Context,
+	})
+
+	if assertion.ListObjectsErrorCode != 0 {
```

Comment:
Do we also want to test when assertion.ErrorCode != 0 that we will return error for ListObjects?

Reply:
I experimented with adding this:

```
if assertion.ErrorCode != 0 {
		fmt.Println(resp)
		require.Error(t, err)
		e, ok := status.FromError(err)
		require.True(t, ok)
		require.Equal(t, assertion.ErrorCode, int(e.Code()))
		return
	}
```

but there is _one_ test that fails: TestMatrixMemory/test_matrix_experimental_false/complete_testing_model/stage_usersets_userset_cond_to_computed_cond/assertion_list_objects_no_cond

It fails correctly, because ListObjects never reads the conditioned tuples, therefore doesn't throw 2000.

I _could_ bring this assertion and skip it just for this PR, but it looks uuugly.

Reply:
In this case, can we ensure all tests that has ErrorCode!=0 also has ListObjectsErrorCode != 0  (except `assertion_list_objects_no_cond`)?